server/remote_cache/digest/digest.go

```diff
@@ -39,25 +39,35 @@ var (
 	// - "blobs/469db13020c60f8bdf9c89aa4e9a449914db23139b53a24d064f967a51057868/39120"
 	// - "blobs/ac/469db13020c60f8bdf9c89aa4e9a449914db23139b53a24d064f967a51057868/39120"
 	// - "uploads/2042a8f9-eade-4271-ae58-f5f6f5a32555/blobs/8afb02ca7aace3ae5cd8748ac589e2e33022b1a4bfd22d5d234c5887e270fe9c/17997850"
-	uploadRegex      = regexp.MustCompile("^(?:(?:(?P<instance_name>.*)/)?uploads/(?P<uuid>[a-f0-9-]{36})/)?blobs/(?P<hash>[a-f0-9]{64})/(?P<size>\\d+)")
-	downloadRegex    = regexp.MustCompile("^(?:(?P<instance_name>.*)/)?blobs/(?P<hash>[a-f0-9]{64})/(?P<size>\\d+)")
-	actionCacheRegex = regexp.MustCompile("^(?:(?P<instance_name>.*)/)?blobs/ac/(?P<hash>[a-f0-9]{64})/(?P<size>\\d+)")
+	uploadRegex      = regexp.MustCompile(`^(?:(?:(?P<instance_name>.*)/)?uploads/(?P<uuid>[a-f0-9-]{36})/)?(?P<blob_type>blobs|compressed-blobs/zstd)/(?P<hash>[a-f0-9]{64})/(?P<size>\d+)`)
+	downloadRegex    = regexp.MustCompile(`^(?:(?P<instance_name>.*)/)?(?P<blob_type>blobs|compressed-blobs/zstd)/(?P<hash>[a-f0-9]{64})/(?P<size>\d+)`)
+	actionCacheRegex = regexp.MustCompile(`^(?:(?P<instance_name>.*)/)?(?P<blob_type>blobs|compressed-blobs/zstd)/ac/(?P<hash>[a-f0-9]{64})/(?P<size>\d+)`)
 )
 
-type InstanceNameDigest struct {
+type ResourceName struct {
 	*repb.Digest
 	instanceName string
+	compressor   repb.Compressor_Value
 }
 
-func NewInstanceNameDigest(d *repb.Digest, instanceName string) *InstanceNameDigest {
-	return &InstanceNameDigest{
+func NewResourceName(d *repb.Digest, instanceName string, compressor repb.Compressor_Value) *ResourceName {
+	return &ResourceName{
 		Digest:       d,
 		instanceName: instanceName,
+		compressor:   compressor,
 	}
 }
```

Comment:
I think it's worth being really really careful with the naming of this struct because it's used *everywhere*. Calling this struct a "ResourceName" feels wrong to me because "resource name" already means something: it's the string name for an uploaded or downloaded digest, like "/blobs/foo/my_cool_digest/123".

My first inclination was to call this just Digest, but i think it's a bit confusing because this Digest would not be the same as repb.Digest (which does not include compression or instanceName).

We could call this struct "Resource", which is about as bad as Digest tbh, but works with ResourceName below.

Reply:
Resolved offline, the reasoning for naming this `ResourceName` is that the original naming of the proto field "resource name" is likely referring to a URN (Uniform Resource Name), and so it makes sense to follow a similar to convention as `*url.URL` in the go stdlib, where the `*url.URL` refers to a struct, while a `string` refers to a serialized version of the struct, but both are called "URL".

Also sticking with `ResourceName` instead of `Resource` to avoid confusion since this thing is a "pointer" to a resource and not the resource itself.